### PR TITLE
Make all API models convert to camelCase

### PR DESCRIPTION
**Pull request by callumforrester**
Previously, with apischema, the API models were serialized with camel case aliasing. For example, converting a Python field called `foo_bar` to a JSON field called `fooBar` and vice versa. This is to comply with the Google JSON style guide. 

https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format 

This PR adds a custom base model with custom config primarily to preserve this change and also to prevent the ingestion of arbitrary JSON alongside a model's known fields, which apischema also did not allow.

### CODE REVIEW

File: src/blueapi/utils/base_model.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+from pydantic import BaseConfig, BaseModel, Extra
+
+
+def _to_camel(string: str) -> str:
```
Comment (DiamondJoseph) [inline src/blueapi/utils/base_model.py:4]: Some words on why this change is required would be useful to prevent it being undone again at some latter point

Comment (callumforrester) [reply src/blueapi/utils/base_model.py:4]: Done in docstring of `BlueapiPlanModelConfig`

**Review by DiamondJoseph — APPROVED** [review]
